src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt

```diff
@@ -64,12 +64,7 @@ class SentencesExtractionService {
   }
 
   fun allSentencesContainType(sentences: MutableList<Sentence>, sentenceType: SentenceType): Boolean {
-    for (sentence in sentences) {
-      if (!sentence.sentenceTypes.contains(sentenceType)) {
-        return false
-      }
-    }
-    return true
+    return sentences.stream().allMatch { sentence -> sentence.sentenceTypes.contains(sentenceType) }
```

Comment:
Replaced with a stream based approach @yasinmustafa  :)

Reply:
In kotlin you dont need to explicitly convert to a stream..s o the above could simply be (`it` is a keyword and refers to the current element in the list without having to explicitly name it):
`return sentences.all { it.sentenceTypes.contains(sentenceType) }`

Reply:
Nice - good learning experience :)